akka-stream/src/main/scala/akka/stream/impl/io/OutputStreamSourceStage.scala

```diff
@@ -107,6 +107,8 @@ final private[stream] class OutputStreamSourceStage(writeTimeout: FiniteDuration
           //assuming there can be no further in messages
           downstreamStatus.set(Canceled)
           dataQueue.clear()
+          // if blocked reading, make sure the take() completes
+          dataQueue.put(ByteString())
```

Comment:
Do you mean the empty `ByteString`? there is a vararg apply, so it will be an empty array of bytes.